x-pack/plugins/alerts/server/saved_objects/partially_update_alert.test.ts

```diff
@@ -0,0 +1,112 @@
+  ISavedObjectsRepository
+>;
+
+describe('partially_update_alert', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  for (const [soClientName, soClient] of Object.entries(getMockSavedObjectClients()))
+    describe(`using ${soClientName}`, () => {
+      test('should work with no options', async () => {
+        soClient.update.mockResolvedValueOnce(MockUpdateValue);
+
+        await partiallyUpdateAlert(soClient, MockAlertId, DefaultAttributes);
+        expect(soClient.update).toHaveBeenCalledWith('alert', MockAlertId, DefaultAttributes, {});
+      });
+
+      test('should work with extraneous attributes ', async () => {
+        const attributes = (InvalidAttributes as unknown) as PartiallyUpdateableAlertAttributes;
+        soClient.update.mockResolvedValueOnce(MockUpdateValue);
+
+        await partiallyUpdateAlert(soClient, MockAlertId, attributes);
+        expect(soClient.update).toHaveBeenCalledWith('alert', MockAlertId, DefaultAttributes, {});
+      });
+
+      test('should handle SO errors', async () => {
+        soClient.update.mockRejectedValueOnce(new Error('wops'));
+
+        await expect(
+          partiallyUpdateAlert(soClient, MockAlertId, DefaultAttributes)
+        ).rejects.toThrowError('wops');
+      });
+
+      test('should handle the version option', async () => {
+        soClient.update.mockResolvedValueOnce(MockUpdateValue);
+
+        await partiallyUpdateAlert(soClient, MockAlertId, DefaultAttributes, { version: '1.2.3' });
+        expect(soClient.update).toHaveBeenCalledWith('alert', MockAlertId, DefaultAttributes, {
+          version: '1.2.3',
+        });
+      });
+
+      test('should handle the ignore404 option', async () => {
+        const err = SavedObjectsErrorHelpers.createGenericNotFoundError();
+        soClient.update.mockRejectedValueOnce(err);
+
+        await partiallyUpdateAlert(soClient, MockAlertId, DefaultAttributes, { ignore404: true });
+        expect(soClient.update).toHaveBeenCalledWith('alert', MockAlertId, DefaultAttributes, {});
+      });
+
+      test('should handle the namespace option', async () => {
+        soClient.update.mockResolvedValueOnce(MockUpdateValue);
+
+        await partiallyUpdateAlert(soClient, MockAlertId, DefaultAttributes, {
+          namespace: 'bat.cave',
+        });
+        expect(soClient.update).toHaveBeenCalledWith('alert', MockAlertId, DefaultAttributes, {
+          namespace: 'bat.cave',
+        });
+      });
+    });
+});
+
+function getMockSavedObjectClients(): Record<
+  string,
+  jest.Mocked<SavedObjectsClientContract | ISavedObjectsRepository>
+> {
+  return {
+    SavedObjectsClientContract: MockSavedObjectsClientContract,
+    // doesn't appear to be a mock for this, but it's basically the same as the above,
+    // so just cast it to make sure we catch any type errors
+    ISavedObjectsRepository: MockISavedObjectsRepository,
+  };
+}
```

Comment:
I might be missing something, but why do we need to support both the client and repo? 🤔 
Looks like we're only using the client, but supporting both adds complexity in both the tests and the implementation.

Reply:
Ah sorry - this is needed for the alert status PR - I copied the code from there, and it needs the repository, because ... well, not quite sure, but API wise, I could only get a repository and not a client (need a kibana system SO "client" to be used in the alert executor task).  We'll see this again when that PR is ready for review, we could look at removing it then.

Reply:
I've changed this to a type which is just the `update()` method in SavedObjectsClient - which will also work for SavedObjectsRepository once we get there.